### PR TITLE
Fixes to the changed image list generation for pushes/PRs

### PR DESCRIPTION
**Pull request by brianhlin**
This PR should fix the CI for pushes to personal forks (for both `main` and any other branch) and still work for PRs and pushes to `main` on this fork. Since this includes #2 , it should build all 3 images that we currently have and once that's merged, we can kick off these tests again and all the build jobs will be skipped.

### CODE REVIEW

File: opensciencegrid/ospool-cm/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,8 @@
-FROM opensciencegrid/software-base:3.6-el7-release
+# Specify the opensciencegrid/software-base image tag
+ARG BASE_OSG_SERIES=3.5
```
Comment (matyasselmeci) [inline opensciencegrid/ospool-cm/Dockerfile:2]:
```suggestion
ARG BASE_OSG_SERIES=3.6
```

**Review by matyasselmeci — APPROVED** [review]
Looks good other than my one comment

**Comment by brianhlin** [comment]
All images were built as expected. Once I merge Mat's suggestion, I expect image builds to be skipped

**Comment by brianhlin** [comment]
And that worked! The CI on my fork built all the images (https://github.com/brianhlin/osg-images/actions/runs/1729441086) because I haven't updated my `main` branch to match this one.

**Pull request merged**